### PR TITLE
Added logs necessary for tracking slot related issues.

### PR DESCRIPTION
**Pull request by abeykoon**
Also fixed warnings in RDBMSMessageStoreImpl due to java  language level.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 1]
```diff
@@ -208,8 +208,10 @@ private Slot getUnassignedSlot(String queueName) throws AndesException {
 			slotToBeAssigned = slotAgent.getUnAssignedSlot(queueName);
 
 			if (log.isDebugEnabled()) {
-				log.debug("Giving a slot from unassigned slots. Slot: " + slotToBeAssigned +
-				          " to queue: " + queueName);
+                if(null != slotToBeAssigned) {
+                    log.debug("Giving a slot from unassigned slots. Slot: "
+                            + slotToBeAssigned + " to queue: " + queueName);
+                }
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Need to fix indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 2]
```diff
@@ -229,7 +231,12 @@ private Slot getOverlappedSlot(String nodeId, String queueName) throws AndesExce
 		synchronized (lockKey.intern()) {
 			//get oldest overlapped slot from database
 			slotToBeAssigned = slotAgent.getOverlappedSlot(nodeId, queueName);
-		}
+        }
+        if (log.isDebugEnabled()) {
+            if (null != slotToBeAssigned) {
+                log.debug(" Giving overlapped slot id=" + slotToBeAssigned.getId() + " queue name= " + queueName);
+            }
+        }
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Need to fix indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 3]
```diff
@@ -381,21 +388,25 @@ public boolean deleteSlot(String queueName, Slot emptySlot, String nodeId) throw
 		long endMsgId = emptySlot.getEndMessageId();
 		long slotDeleteSafeZone = getSlotDeleteSafeZone();
 		if (log.isDebugEnabled()) {
-			log.debug("Trying to delete slot. safeZone= " + getSlotDeleteSafeZone() +
-			          " startMsgID: " + startMsgId);
+			log.debug("Trying to delete slot. safeZone= " + getSlotDeleteSafeZone()
+                    + " startMsgID: " + startMsgId);
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Need to fix indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 4]
```diff
@@ -381,21 +388,25 @@ public boolean deleteSlot(String queueName, Slot emptySlot, String nodeId) throw
...
 		}
 		if (slotDeleteSafeZone > endMsgId) {
 			String lockKey = nodeId + SlotManagerClusterMode.class;
 			synchronized (lockKey.intern()) {
 				slotAgent.deleteSlot(nodeId, queueName, startMsgId, endMsgId);
+                if (log.isDebugEnabled()) {
+                    log.debug(" Deleted slot id = " + emptySlot.getId() + " queue name = " + queueName);
+                }
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Need to fix indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 5]
```diff
@@ -517,7 +528,7 @@ public void shutDownSlotManager() {
 				slot.setAnOverlappingSlot(true);
 
 				if (log.isDebugEnabled()) {
-					log.debug("Marked already assigned slot as an overlapping slot. Slot= " + slot);
+                    log.debug("Marked already assigned slot as an overlapping slot. Slot= " + slot.getId());
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Need to fix indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java (hunk 2 above)
Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Fixed in updated PR

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java (hunk 5 above)
Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Fixed in updated PR

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java (hunk 3 above)
Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Fixed in updated PR

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java (hunk 4 above)
Comment (abeykoon) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: Updated PR with correct indentation

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java [hunk 6]
```diff
@@ -200,20 +200,22 @@ private Slot getFreshSlot(String queueName, String nodeId) throws AndesException
...
+        String lockKey = queueName + SlotManagerClusterMode.class;
+        synchronized (lockKey.intern()) {
+            //get oldest unassigned slot from database
+            slotToBeAssigned = slotAgent.getUnAssignedSlot(queueName);
+
+            if (log.isDebugEnabled()) {
+                if (null != slotToBeAssigned) {
+                    log.debug("Giving a slot from unassigned slots. Slot: "
+                            + slotToBeAssigned + " to queue: " + queueName);
+                }
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java]: What is the change you have done here?

**Comment by abeykoon** [comment]
closing due to formatting issues. Should redo via another pull request.

**Pull request closed without merging**